ui/apps/app-center/src/extensions/integrations-widget.tsx

```diff
@@ -19,38 +20,49 @@ const { Box, ICWidgetCard, ErrorLoader } = DS;
 const ICWidget: React.FC<RootComponentProps> = props => {
   const { t } = useTranslation('app-integration-center');
 
+  const { worldConfig } = props;
+
   const loginQuery = useGetLogin();
 
   const isLoggedIn = React.useMemo(() => {
     return !!loginQuery.data.pubKey;
   }, [loginQuery.data]);
 
-  const defaultAppsNamesNormalized = props.worldConfig?.defaultApps.map(app => {
-    if (typeof app === 'string') {
-      return {
-        name: app,
-      };
+  const availableIntegrationsReq = useGetAllIntegrationsIds(isLoggedIn);
+
+  const defaultIntegrations = [].concat(
+    worldConfig.defaultApps,
+    worldConfig.defaultWidgets,
+    [worldConfig.homepageApp],
+    [worldConfig.layout],
+  );
+
+  const integrationIdsNormalized = React.useMemo(() => {
+    if (availableIntegrationsReq.data?.integrationIds) {
+      return availableIntegrationsReq.data?.integrationIds.map(integrationId => {
+        return { id: integrationId };
+      });
     }
-    return app;
-  });
+    return worldConfig.defaultApps.map(integrationName => {
+      return { name: integrationName };
+    });
+  }, [availableIntegrationsReq.data, worldConfig.defaultApps]);
 
   const installedAppsReq = useGetAllInstalledApps(isLoggedIn);
-  const installedIntegrationsInfoReq = useGetIntegrationsInfo(installedAppsReq.data);
+  const integrationsInfoReq = useGetIntegrationsInfo(integrationIdsNormalized);
 
-  // select default apps from list of installed apps
-  const filteredDefaultApps = installedIntegrationsInfoReq.data?.getIntegrationInfo.filter(app => {
-    if (defaultAppsNamesNormalized?.some(defaultApp => defaultApp.name === app.name)) {
+  // select default apps from list of apps
+  const filteredDefaultApps = integrationsInfoReq.data?.getIntegrationInfo.filter(app => {
+    if (worldConfig.defaultApps?.some(defaultApp => defaultApp === app.name)) {
       return app;
     }
   });
   // select user installed apps from list of installed apps
-  const filteredInstalledApps = installedIntegrationsInfoReq.data?.getIntegrationInfo.filter(
-    app => {
-      if (!defaultAppsNamesNormalized?.some(defaultApp => defaultApp.name === app.name)) {
-        return app;
-      }
-    },
-  );
+  const filteredInstalledApps = integrationsInfoReq.data?.getIntegrationInfo.filter(app => {
+    if (defaultIntegrations?.some(defaultApp => defaultApp !== app.name)) {
```

Comment:
I think this check is redundant because an app can either be in defaultIntegrations or in installedAppReq.
What i would do here is a `partition` thus doing only one iteration over the `integrationsInfoReq.data?.getIntegrationInfo`

something like:
```
const {filteredDefault, filteredInstalled} = integrationsInfoReq.data?.getIntegrationInfo.reduce((acc, app) => {
    if(defaultIntegrations.includes(app.name) {
      acc.filteredDefault.push(app)
    } else {
     // is probably installed but you can also check to be sure
     acc.filteredInstalled.push(app);
    }
    return acc;
}, {filteredDefault: [], filteredInstalled: []});
```

Reply:
ok, I can change it to this pattern